src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ExpressionTreeCallRewriter.cs

```diff
@@ -221,7 +217,7 @@ private Expr GenerateLambda(ExprCall pExpr)
         {
             // We always call Lambda(body, arrayinit) where the arrayinit 
             // is the initialization of the parameters.
-            return Visit(pExpr.OptionalArguments.asLIST().OptionalElement);
+            return Visit((pExpr.OptionalArguments as ExprList).OptionalElement);
```

Comment:
I started with `()` casts in cases where it seemed appropriate, but decided that an NRE perhaps isn't that much harder to trace than a ICE when the cause is a bad downcast, and it made the mechanics of the change a lot easier.